src/lib.rs

```diff
@@ -133,7 +142,9 @@ where
 pub struct BusyTimeout;
 
 pub trait BusyWait {
-    fn poll_wait(&mut self) -> Result<(), BusyTimeout>;
+    #[cfg_attr(not(feature = "async"), remove_async_await::remove_async_await)]
+    #[cfg_attr(feature = "async", allow(async_fn_in_trait))]
+    async fn poll_wait(&mut self) -> Result<(), BusyTimeout>;
```

Comment:
I'm not sure the best route to take with this is. Currently, having async functions in traits is not recommended and causes a warning because it results in the Future not having auto traits such as Send implemented on it.

Reply:
agreed, when async is enabled, BusyWait should not exist